core/trino-spi/src/test/java/io/trino/spi/type/TestingTypeManager.java

```diff
@@ -32,9 +35,39 @@
 public class TestingTypeManager
         implements TypeManager
 {
-    private static final List<Type> TYPES = ImmutableList.of(BOOLEAN, BIGINT, DOUBLE, INTEGER, VARCHAR, VARBINARY, TIMESTAMP_MILLIS, TIMESTAMP_WITH_TIME_ZONE, DATE, ID, HYPER_LOG_LOG);
-
-    private final TypeOperators typeOperators = new TypeOperators();
+    private static final TypeOperators typeOperators = new TypeOperators();
+    private static final List<Type> TYPES = ImmutableList.of(
+            BOOLEAN,
+            BIGINT,
+            DOUBLE,
+            INTEGER,
+            VARCHAR,
+            VARBINARY,
+            TIMESTAMP_MILLIS,
+            TIMESTAMP_WITH_TIME_ZONE,
+            DATE,
+            ID,
+            HYPER_LOG_LOG,
+            // types below are used in BenchmarkPartitionedOutputOperator
+            createMapType(BIGINT, BIGINT),
+            createMapType(createDecimalType(MAX_SHORT_PRECISION + 1), createDecimalType(MAX_SHORT_PRECISION + 1)),
+            createMapType(INTEGER, INTEGER),
```

Comment:
does it make sense to test all these map variants? Would it be possible to pick 3-4 variants that make sense?

For example
* execution is mostly same for flat plain maps, e.g: `createMapType(INTEGER, INTEGER), createMapType(SMALLINT, SMALLINT)`.
* you could pick one set of nested map.